tools/snp.sh

```diff
@@ -1073,50 +1100,140 @@ generate_snp_expected_measurement() {
     { >&2 echo -e "sev-snp-measure return value is empty"; return 1; }
   echo ${measurement}
 }
-
 attest_guest() {
-  local cpu_code_name=$(get_cpu_code_name)
+    local cpu_code_name=$(get_cpu_code_name)
+
+    # Install the sev-guest module
+    ssh_guest_command "sudo insmod /lib/modules/*/kernel/drivers/virt/coco/sev-guest/sev-guest.ko >/dev/null 2>&1 || true"
+
+    # Request and display the snp attestation report with random data
+    if [ "$SVSM" = true ]; then
+      ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random"
+    else
+      ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random --vmpl 3"
+    fi
+    ssh_guest_command "./snpguest display report attestation-report.bin"
+
+    # Retrieve ark, ask, vcek (saved in ./certs)
+    ssh_guest_command "./snpguest fetch ca pem ${cpu_code_name} ."
+    ssh_guest_command "./snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin"
+
+    # Verifies that ARK, ASK and VCEK are all properly signed
+    ssh_guest_command "./snpguest verify certs ."
 
-  # Install the sev-guest module
-  ssh_guest_command "sudo insmod /lib/modules/*/kernel/drivers/virt/coco/sev-guest/sev-guest.ko >/dev/null 2>&1 || true"
+    # Verifies the attestation-report trusted compute base matches vcek
+    ssh_guest_command "./snpguest verify attestation . attestation-report.bin"
 
-  # Request and display the snp attestation report with random data
-  ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random"
-  ssh_guest_command "./snpguest display report attestation-report.bin"
+    # Use appropriate measurement function based on SVSM
+    if [ "$SVSM" = true ]; then
+      local expected_measurement=$(generate_snp_expected_measurement)
+      echo -e "\nExpected Measurement (sev-snp-measure): ${expected_measurement}"
+    else
+      local expected_measurement=$(generate_svm_expected_measurement)
```

Comment:
Should we name the function `generate_svsm_expected_measurement` instead of `generate_svm_expected_measurement`